src/server/container.ts

```diff
@@ -124,11 +124,21 @@ export default class Container {
     private saveCodeFiles = (code: string, includeFileData: Buffer | null): void => {
         // Create data directory and save code from request
         console.log(`${this.logPrefix}Saving code to ${this.codeHostPath}`);
-        if (!fs.existsSync(this.dataHostPath)) fs.mkdirSync(this.dataHostPath);
-        fs.writeFileSync(this.codeHostPath, code);
+        fs.access(this.dataHostPath, fs.constants.F_OK, (errPathExists) => {
```

Comment:
I'd rename `errPathExists` to something like `accessError`, since the "PathExists" part could imply that the path does exist